partitioner/src/main/java/io/confluent/connect/storage/partitioner/FieldPartitioner.java

```diff
@@ -17,55 +17,76 @@
 package io.confluent.connect.storage.partitioner;
 
 import org.apache.kafka.connect.data.Schema;
-import org.apache.kafka.connect.data.Schema.Type;
 import org.apache.kafka.connect.data.Struct;
 import org.apache.kafka.connect.sink.SinkRecord;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-
+import java.util.List;
 import java.util.Map;
 
 import io.confluent.connect.storage.common.StorageCommonConfig;
 import io.confluent.connect.storage.errors.PartitionException;
 
 public class FieldPartitioner<T> extends DefaultPartitioner<T> {
   private static final Logger log = LoggerFactory.getLogger(FieldPartitioner.class);
-  private String fieldName;
+  private List<String> fieldNames;
 
   @Override
   public void configure(Map<String, Object> config) {
-    fieldName = (String) config.get(PartitionerConfig.PARTITION_FIELD_NAME_CONFIG);
-    partitionFields = newSchemaGenerator(config).newPartitionFields(fieldName);
+    fieldNames = (List<String>) config.get(PartitionerConfig.PARTITION_FIELD_NAME_CONFIG);
     delim = (String) config.get(StorageCommonConfig.DIRECTORY_DELIM_CONFIG);
+
+    StringBuilder format = new StringBuilder();
+    for (String field : fieldNames) {
+      format.append(field + delim);
+    }
+
+    if (format.length() > 0) {
+      format.delete(format.length() - delim.length(), format.length());
+    }
+
+    partitionFields = newSchemaGenerator(config).newPartitionFields(format.toString());
   }
 
   @Override
   public String encodePartition(SinkRecord sinkRecord) {
     Object value = sinkRecord.value();
     Schema valueSchema = sinkRecord.valueSchema();
+    StringBuilder partition = new StringBuilder();
```

Comment:
Instead of building the joined set of partition fields manually, I'd suggest adding all the fields in a `List` and then using `org.apache.kafka.common.utils.Utils.join` to return the final `String`.